scipy/integrate/tests/test_quadrature.py

```diff
@@ -265,32 +266,34 @@ def test_zero_len_y(self):
 
 
 class TestTrapezoid:
-    def test_simple(self):
-        x = np.arange(-10, 10, .1)
-        r = trapezoid(np.exp(-.5 * x ** 2) / np.sqrt(2 * np.pi), dx=0.1)
+    @array_api_compatible
+    def test_simple(self, xp):
+        x = xp.arange(-10, 10, .1)
+        r = trapezoid(xp.exp(-.5 * x ** 2) / xp.sqrt(2 * xp.asarray(xp.pi)), dx=0.1)
         # check integral of normal equals 1
         assert_allclose(r, 1)
```

Comment:
we want to use `xp_assert_close` from `scipy._lib._array_api` instead of `assert_allclose` from `np.testing` here. It comes with additional strictness checks, the most relevant one here being a check that the namespaces match. You may have to grapple with whether we are expecting arrays or scalars, too, after switching to the new assertions.

(this could probably be made clearer in our array API dev docs, though!)